src/main/java/io/supertokens/session/Session.java

```diff
@@ -423,11 +430,14 @@ public static SessionInformationHolder refreshSession(AppIdentifier appIdentifie
             }
         }
 
-        return refreshSessionHelper(refreshTokenInfo.tenantIdentifier, main, refreshToken, refreshTokenInfo,
+        return refreshSessionHelper(
```

Comment:
the function for getInfoFromRefreshToken should also store the appIdHash and do a check similar to in the access token

Reply:
Done